editoast/src/chartos/mod.rs

```diff
@@ -1,102 +1,139 @@
-    "detectors",
-    "buffer_stops",
-    "routes",
-    "operational_points",
-    "catenaries",
-    "lpv_panels",
-    "errors",
-];
+use self::layer_cache::{
+    count_tiles, get_cache_tile_key, get_layer_cache_prefix, get_tiles_to_invalidate,
+    get_view_cache_prefix, Tile,
+};
+use self::redis_utils::{delete, keys};
 
-/// Invalidate a zone for all chartos layers
-/// If the zone is invalide nothing is done
-pub async fn invalidate_zone(
+/// Invalidates layer cache for a specific infra and view if provided
+///
+/// # Arguments
+///
+/// * `redis_pool` - Pool to use to connect to the redis
+/// * `infra_id` - Infra to on which the layer must be invalidated
+/// * `layer_name` - Layer to invalidate
+/// * `view_name` - Specific view to invalidate, if not provided all layer's views are invalidated
+///
+/// Returns the number of deleted keys
+async fn invalidate_full_layer_cache(
+    redis_pool: &RedisPool,
     infra_id: i32,
-    chartos_config: &ChartosConfig,
-    zone: &InvalidationZone,
-) {
-    if !zone.is_valid() {
-        return;
-    }
+    layer_name: &str,
+    view_name: Option<&str>,
+) -> Result<u64, RedisError> {
+    let prefix: String = view_name.map_or(get_layer_cache_prefix(layer_name, infra_id), |view| {
+        get_view_cache_prefix(layer_name, infra_id, view)
+    });
+    let matching_keys = keys(redis_pool, &format!("{prefix}.*")).await?;
+    let number_of_deleted_keys = delete(redis_pool, matching_keys).await?;
+    Ok(number_of_deleted_keys)
+}
 
-    for layer in LAYERS {
-        invalidate_layer_zone(infra_id, layer, zone, chartos_config).await;
+/// Invalidates cache for specific
+///
+/// # Arguments
+///
+/// * `redis_pool` - Pool to use to connect to the redis
+/// * `tiles_to_invalidate` - View cache prefix to tiles
+async fn invalidate_cache_tiles(
+    redis_pool: &RedisPool,
+    tiles_to_invalidate: HashMap<String, Vec<Tile>>,
+) -> Result<u64, RedisError> {
+    let mut keys_to_delete: Vec<String> = Vec::new();
+    for (view_cache_prefix, tiles) in tiles_to_invalidate {
+        for tile in tiles {
+            keys_to_delete.push(get_cache_tile_key(&view_cache_prefix, &tile));
+        }
     }
+    let number_of_deleted_keys = delete(redis_pool, keys_to_delete).await?;
+    Ok(number_of_deleted_keys)
 }
 
-/// Invalidate all chartos layers
-pub async fn invalidate_all(infra_id: i32, chartos_config: &ChartosConfig) {
-    for layer in LAYERS {
-        invalidate_layer(infra_id, layer, chartos_config).await;
+/// Invalidates a infra specific layer zone
+///
+/// # Arguments
+///
+/// * `redis_pool` - Pool to use to connect to the redis
+/// * `infra_id` - Infra to on which the layer must be invalidated
+/// * `layer_name` - Layer on which invalidation must be done
+/// * `zone` - Zone to invalidate
+async fn invalidate_layer_zone(
+    redis_pool: &RedisPool,
+    infra_id: i32,
+    layer_name: &str,
+    zone: &InvalidationZone,
+) -> Result<(), RedisError> {
+    let max_tiles = 120;
```

Comment:
Can we add this in the the configuration file (or as a cli arg)?
Also the default value was set to `250 000`.